lib/env_compare/cli.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'erb'
+require 'launchy'
+require 'securerandom'
+require 'pry'
+require 'thor'
+
+module EnvCompare
+  class CLI < Thor
+    attr_accessor :envs
+
+    desc 'version', 'list version'
+    def version
+      puts EnvCompare::VERSION
+    end
+
+    desc 'list ENVIRONMENT', 'List environment variables for a single environment'
+
+    def list(environment)
+      output = `heroku config -a #{environment}`
+      puts output
+    end
+
+    option :theme, type: :string, default: 'dark', desc: 'Theme to use'
+    option :all, type: :boolean, default: false, desc: 'Display all ENV variables, including matches'
+    desc 'diff heroku-app-name-1 heroku-app-name-2', 'Compare environment variables for 2+ environments'
+    def diff(*envs)
+      return list(envs.first) if envs.size == 1
+
+      heroku_results(envs)
+
+      # https://stackoverflow.com/a/5462069/2892779
+      html =
+        theme(options[:theme]).
+          result(
+            ::EnvCompare::NameSpace.new(data: calculated_output(envs), headers: ['KEY', envs].flatten).get_binding
+          )
+
+      save_and_open_file(html)
+    end
+
+    private
+
+    def all_keys
+      @all_keys ||= Set.new
+    end
+
+    def calculated_output(envs)
+      all_keys.sort.map do |key|
+        results = envs.map { |env| heroku_results[env][key] }
+        next if results.uniq.size == 1 && !options[:all]
+
+        [key, *results]
+      end.compact
+    end
+
+    def heroku_results(envs = nil)
+      @heroku_results ||=
+        envs.each_with_object({}) do |env, obj|
+          obj[env] = {}
+          lines = `heroku config -a #{env}`.split("\n").drop(1)
+
+          lines.each do |line|
+            key, _, value = line.partition(':')
```

Comment:
I think we can do `key, value = line.split(':')` and avoid the unnecessary `_` since `strip` is being used regardless?

Reply:
This was a copy/paste. This goes away in my next PR

Reply:
Sweet!